refactor(ast)!: rename `Expression::without_parentheses`

overlookmotel opened the pull request:
Rename `without_parenthesized` to `without_parentheses`. Both are hard-to-spell words, but "without parenthesized" does not make sense.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5448?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5448** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5448?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5447** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5447?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5446** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5446?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5445** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5445?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
Diff is large, but it's only renaming that 1 function.

overlookmotel commented:
Ooh I like this autofix!

DonIsaac commented:
This should be marked as a breaking change since this rename affects our public API.

overlookmotel commented:
Good point. I've added `!` to the PR name.

Boshen commented:
`without_parenthesized` means `without parenthesized expression`.

overlookmotel committed:
refactor(ast)!: rename `Expression::without_parentheses` (#5448)

Rename `without_parenthesized` to `without_parentheses`. Both are hard-to-spell words, but "without parenthesized" does not make sense.
